src/lib/utils/async.js

```diff
@@ -23,3 +24,22 @@ export const retry = (asyncFn, retries = 5, interval = 0) =>
       ),
     )
     .toPromise()
+
+// eslint-disable-next-line require-await
+export const promisifyGun = async callback =>
+  new Promise((resolve, reject) => {
+    const onAck = once(ack => {
```

Comment:
promise can be resolved or rejected  only once so i  dont think using once required

Reply:
We have here an additional logic including instantiation if the Error object (which includes capturing a large stack object). So to optimize app better to wrap it to once that guaranteed the logic and new Error will be called only if they're needed (on the first call when res/rej actually could change promise state)